### ISSUE TITLE
feature/Support for miri?

### ISSUE THREAD

**Issue by NobodyXu** [nextest-rs/nextest#181]
Hi

First, thank you for this awesome project!

Using `nextest` in medium size projects such as [`compact_str`] and [`bytes`] result in significant speedup.

It will be great if we can also run miri using `nextest`, as miri is very slow at running tests and is single-threaded.

[`compact_str`]: https://github.com/ParkMyCar/compact_str
[`bytes`]: https://github.com/tokio-rs/bytes

**Comment by eddyb** [comment]
Another miri usecase (described to me by @BoxyUwU) where nextest could come in handy, is having many tests where some of them are leaking, which makes it hard to know what test it was (at least in the absence of tagging the allocation with the thread it originated on and mapping that back).

Looking at [how `cargo miri` describes its own operation](https://github.com/rust-lang/miri/blob/4756be8aeebedd6d12f36613cf06e4dfa8aa896b/cargo-miri/bin.rs#L1097-L1103), my understanding is that:
* it mostly treats `cargo test` as a strange flavor of `cargo run`
* it uses `RUSTC_WRAPPER` and [`CARGO_TARGET_<triple>_RUNNER`](https://doc.rust-lang.org/cargo/reference/config.html#targettriplerunner) to customize both "build" and "run" aspects of `cargo run` (and therefore `cargo test` as well)
* the actual "executable" that Cargo just assumes is in the native format for the target, is instead a JSON file describing all the information Cargo passes to the "build" step (like `rustc` flags)

From a quick check, `cargo miri test --lib -- ...` behaves identically to `cargo test --lib -- ...` (with e.g. `--list`), so anything build around invoking `cargo test` would also work with `cargo miri test`.

For example, I came up with this quick hack to help @BoxyUwU out:
```sh
cargo --quiet test --lib -- --list | sed 's/: test$//' | xargs -n1 cargo miri test --lib -- --exact
```
But that's much more fragile than what nextest does *and* AFAIK nextest invokes the test runner executable itself, so nextest going through `cargo miri test` doesn't sound promising.

<hr/>

Instead, since [nextest respects `CARGO_TARGET_<triple>_RUNNER`](https://nexte.st/book/target-runners.html), it should be possible to do this from the `cargo-miri` side, specifically adding a `cargo miri nextest` subcommand (it would go in [this part of the source code](https://github.com/rust-lang/miri/blob/4756be8aeebedd6d12f36613cf06e4dfa8aa896b/cargo-miri/bin.rs#L555-L586)) that behaves like `cargo miri test` but running `cargo nextest` instead of `cargo test` (not sure how the `cargo nextest` subcommand passing would work, but to `cargo-miri` it might just be freeform enough to not matter).

In fewer words, I agree with this:
> I think support for nextest should be added to cargo miri itself

and also think it would take very little effort (assuming no weird Cargo features get in the way)

(@oli-obk, @RalfJung, or someone else who works on miri may also want to chime in)

**Comment by RalfJung** [comment]
Your summary of `cargo miri` operations sounds accurate. :)

**Comment by sunshowers** [comment]
Just released cargo-nextest 0.9.27 with support for Miri. https://nexte.st/book/miri.html

**Comment by RalfJung** [comment]
Awesome. :)
The next nightly that ships Miri should have the required support for this. (The currently latest nightly with Miri does not have it yet.)
